packages/client/src/components/Canvas/resources.ts

```diff
@@ -0,0 +1,56 @@
+class ResourcesHandler {
+  private resourceCache: { [key: string]: HTMLImageElement | boolean } = {};
+
+  //private loading: string[] = [];
+
+  private readyCallbacks: (() => void)[] = [];
+
+  public load(urlOrArr: string | string[]) {
+    if (Array.isArray(urlOrArr)) {
+      urlOrArr.forEach(url => {
+        this._load(url);
+      });
+    } else {
+      this._load(urlOrArr);
+    }
+  }
+
+  private _load(url: string) {
+    if (this.resourceCache[url]) {
+      return this.resourceCache[url];
+    } else {
+      const img = new Image();
+      img.onload = () => {
+        this.resourceCache[url] = img;
+        if (this.isReady()) {
+          this.readyCallbacks.forEach(func => {
+            func();
+          });
+        }
+      };
+      this.resourceCache[url] = false;
+      img.src = url;
+    }
+  }
+
+  public get(url: string) {
+    return this.resourceCache[url] as HTMLImageElement;
+  }
+
+  public isReady() {
+    let ready = true;
+    for (const k in this.resourceCache) {
```

Comment:
Мне кажется, нет необходимости перебирать все ключи, и можно упростить:
```suggestion
    public isReady() {
      for (const k in this.resourceCache) {
        if (!this.resourceCache[k]) {
          return false;
        }
      }
      return true;
    }
```